dotCMS/src/integration-test/java/com/dotcms/contenttype/test/ContentTypeTest.java

```diff
@@ -0,0 +1,122 @@
+import com.dotcms.datagen.TestDataUtils;
+import io.vavr.control.Try;
+
+public class ContentTypeTest extends ContentTypeBaseTest {
+
+
+
+    /**
+     * This tests that calling type.fields() returns valid fields and that calling type.fields(Class
+     * clazz) returns a subset of those fields
+     * 
+     * @throws Exception
+     */
+
+    @Test
+    public void test_that_fields_of_type_are_returned() throws Exception {
+        ContentType type = TestDataUtils.getFormLikeContentType();
+
+        assert (!type.fields().isEmpty());
+        assert (type.fields().size() == type.fieldMap().size());
+        assert (!type.fields(ConstantField.class).isEmpty());
+        assert (type.fields(ConstantField.class).size() == 2);
+
+    }
+
+    /**
+     * This tests that calling type.fields(TextField.class) returns the same subset of fields as does
+     * calling type.fields(ImmutableTextField.class)
+     * 
+     * @throws Exception
+     */
+
+    @Test
+    public void test_that_immutable_fields_work_the_same_as_normal_fields_returned() throws Exception {
+
+        ContentType type = TestDataUtils.getFormLikeContentType();
+
+        assert (!type.fields().isEmpty());
+        assert (type.fields().size() == type.fieldMap().size());
+
+        List<Field> constants = type.fields(ConstantField.class);
+        List<Field> immutableConstants = type.fields(ImmutableConstantField.class);
+        assert (!constants.isEmpty());
+        assert (!immutableConstants.isEmpty());
+
+
+
+        // make sure that getting the field by Type or by its immutable type works the same
+        assert (constants.size() == immutableConstants.size());
+        assertEquals(constants, immutableConstants);
+
+    }
+
+
+
+    /**
+     * This tests that the changing of the code in the ContentType to return fields works the same as
+     * the old way we used to do it
+     * 
+     * @throws Exception
+     */
+
+    @Test
+    public void test_that_it_works_like_the_old_way() throws Exception {
+        ContentType type = TestDataUtils.getWidgetLikeContentType();
+
+        assert (!type.fields().isEmpty());
+        assert (type.fields().size() == type.fieldMap().size());
+        assert (!type.fields(ConstantField.class).isEmpty());
+
+
+        List<Field> newWayfields = type.fields(ConstantField.class);
+
+        List<Field> oldWayFields = oldWayOfGettingFields(type, ConstantField.class);
+
+
+        // make sure that getting the field by Type or by its immutable type works the same
+        assert (oldWayFields.size() > 0);
+        assert (newWayfields.size() > 0);
+        assertEquals(oldWayFields, newWayfields);
+
+
+        newWayfields = type.fields(TextField.class);
+
+        oldWayFields = oldWayOfGettingFields(type, TextField.class);
+
+
+        // make sure that getting the field by Type or by its immutable type works the same
+        assert (oldWayFields.size() > 0);
+        assert (newWayfields.size() > 0);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5449504)